File: dart/fake_world/lib/warrior/fighter.dart
import 'package:fake_world/warrior/human.dart';

class Fighter extends Human {
  Fighter({
    required super.name,
    super.dexterity,
    super.strength,
    super.armour,
    super.moxie,
    super.coins,
    super.health,
    super.enemy,
  });
}

File: kotlin/FakeWorld/src/warier/Fighter.kt
package warier

/**
 * Create a Fighter which is a Human
 */
class Fighter : Human {
    /**
     * Create Fighter with random abilities
     *
     * @param name name of the Fighter
     */
    constructor(name: String?) : super(name)

    /**
     * Create a Fighter with given abilities
     *
     * @param name      name of the Fighter
     * @param strength  strength
     * @param dexterity dexterity
     * @param armour    armour
     * @param moxie     moxie
     * @param coins     coins
     * @param health    health
     * @param enemy     an Elf enemy
     */
    constructor(
        name: String?,
        strength: Int,
        dexterity: Int,
        armour: Int,
        moxie: Int,
        coins: Int,
        health: Int,
        enemy: Elf,
    ) : super(name, strength, dexterity, armour, moxie, coins, health, enemy)
}
